Add roles

pascalaubry opened the pull request:
Big PR that implements https://github.com/Sharly-Chess/sharly-chess/blob/feature/roles/docs/sandbox/authentication-and-roles.md
A few screenshots below, I probably need to explain you a few things before you start reviewing...
![image](https://github.com/user-attachments/assets/c5974e08-9f3e-4052-8846-489c41dfe557)
![image](https://github.com/user-attachments/assets/89fc74c5-898c-4fc5-9875-a196617a94a8)
![image](https://github.com/user-attachments/assets/daab68b8-b291-4de3-b1b2-1ea7c7d9a9e3)
![image](https://github.com/user-attachments/assets/5e394773-bb15-406b-afd8-9da0d636aa17)
![image](https://github.com/user-attachments/assets/78765875-cb92-49d4-828d-5820845f80ab)
![image](https://github.com/user-attachments/assets/d14858f5-e64a-45bd-a5a2-95805f741ad2)

pascalaubry committed:
Added role classes

pascalaubry committed:
Merged dev

pascalaubry committed:
Merge branch 'dev' into feature/roles

pascalaubry committed:
Moved m024_add_roles.py to m025_add_roles.py

pascalaubry committed:
Translated strings

pascalaubry committed:
Added the authentication manager

pascalaubry committed:
Merge branch 'dev' into feature/roles

pascalaubry committed:
Fixed Pycharm exception on Ctrl-C

pascalaubry committed:
Merge branch 'dev' into feature/roles

pascalaubry committed:
Merge branch 'dev' into feature/roles

pascalaubry committed:
Fixed m028_add_auth.py

pascalaubry committed:
Added Access entries to the event menu (not finished yet)
Fixed stored objects and migration
Added the user controller

pascalaubry committed:
Merge branch 'dev' into feature/roles

pascalaubry committed:
Simplify access objects
Added the computer controller
Added empty access pages

pascalaubry committed:
Fixed modal IDs

pascalaubry committed:
Added accounts and computer pages

pascalaubry committed:
Simplify computer.matches()

pascalaubry committed:
Fixed FR translation

pascalaubry committed:
Fixed typo

pascalaubry committed:
Added event-free access entities
Simplified the auth manager
Fixed typo

pascalaubry committed:
Moved tab_role_permission_cells.html
Fixed clear_computer_cache()

pascalaubry committed:
Merged dev

pascalaubry committed:
Fixed double m028 migration

pascalaubry committed:
Merged dev

pascalaubry committed:
Added the Profile nav link
Added account authentication

pascalaubry committed:
Merged dev

pascalaubry committed:
Added page titles

pascalaubry committed:
Settings page title
Settings Access control

pascalaubry committed:
Settings Access control

pascalaubry committed:
Events and tournament cards access control
Title on events pages

pascalaubry committed:
Database sources access control

pascalaubry committed:
Event config access control

pascalaubry committed:
Computers and accounts access control

pascalaubry committed:
Merge branch 'dev' into feature/roles

pascalaubry committed:
Merged dev

pascalaubry committed:
Fixed permissions modal

pascalaubry committed:
Tournaments access control (start)

pascalaubry committed:
Merge branch 'dev' into feature/roles

pascalaubry committed:
Tournaments access control (continue)

pascalaubry committed:
Merged dev

pascalaubry committed:
Tournaments access control (end)

pascalaubry committed:
Merged dev

pascalaubry committed:
Players and check-in access control

pascalaubry committed:
Merge branch 'dev' into feature/roles

pascalaubry committed:
Merged dev

pascalaubry committed:
Pairings access control (not finished)

pascalaubry committed:
Merged dev

pascalaubry committed:
Pairings access control (finished)

pascalaubry committed:
Prizes access control (finished)

pascalaubry committed:
Print access control (finished)

pascalaubry committed:
FFE upload access control

pascalaubry committed:
Merged dev

pascalaubry committed:
Merged dev

pascalaubry committed:
Added translations
Added new screen views (not finished yet)

pascalaubry committed:
Merged dev
Screens access control (admin part)
Hide the screens on the admin part by default to make all the screen types visible
Renamed sub_nav to sub_nav_tab in event_layout.html

pascalaubry committed:
Merged dev

pascalaubry committed:
Fixed strings
Fixed test in swiss_pairing_buttons.html

pascalaubry committed:
Fixed admin_screens_show_details on screens/tab.html

pascalaubry committed:
Fixed unknown computers permissions

pascalaubry committed:
Added 'user' display controllers and rotators tabs

pascalaubry committed:
Added 'user' screens tabs

pascalaubry committed:
Fixed exit button link

pascalaubry committed:
Removed the old login stuff
Fixed results and illegal moves entry

pascalaubry committed:
Restored the screen.type == 'input' restriction

pascalaubry committed:
Restored the screen.type == 'input' restriction

pascalaubry committed:
Removed no more used user files

pascalaubry committed:
Completed profile modal
Added refresh listeners
Fixed check-in control on the Pairings tab

pascalaubry committed:
Updated the QR code to fit to the dark background

pascalaubry committed:
Moved the old home page to /admin/home
Removed unused user controllers
Restricted the visualisation of the config to arbiters

pascalaubry committed:
Fixed user controllers

pascalaubry committed:
Completely removed admin_auth()

pascalaubry committed:
Fixed FR strings

pascalaubry committed:
Shortened the navigation titles

pascalaubry committed:
Improved the locale selection

pascalaubry committed:
Improved the home page

pascalaubry committed:
Improved the home page

pascalaubry committed:
Improved the navigation for non-arbiters

pascalaubry committed:
Let organizers manage deputy chief arbiters

pascalaubry committed:
Added execution modes

pascalaubry committed:
Removed update_password
Fixed refresh issue when no tab available

pascalaubry committed:
Merged dev

Molrn commented on src/data/auth/roles.py:
```diff
@@ -0,0 +1,277 @@
+                raise ValueError(f'role={self}')
+
+
+class Role(IntEnum):
```
Using big enums like this to represent complex structures is something we've been trying to remove from the project. It would instead be much better to use `class Role(IdentifiableEntity)` and an entity manager

pascalaubry replied:
I leave this one open since I will need you for it.

Molrn commented on src/data/auth/mode.py:
```diff
@@ -0,0 +1,140 @@
+from data.auth.roles import Role
+
+
+class Mode(IntEnum):
```
if at least modes could be in an individual PR it would be good. seems doable to isolate this code
PS: IntEnum -> IdentifiableEntity

pascalaubry replied:
I leave this one open since I will need you for it.

Molrn commented on src/database/sqlite/event/migrations/m031_add_roles.py:
```diff
@@ -0,0 +1,81 @@
+from data.auth.mode import Mode
+from database.sqlite.event.event_store import (
+    anonymous_stored_account,
+    localhost_stored_computer,
+    unknown_stored_computer,
+)
```
No project import in a migration for stability reasons. If you need to use constants, you redeclare them as they are in the project at the moment of the migration

pascalaubry replied:
I understand but I could not just add constants so I added a `_post_migrate()` to `MigrationDatabase`.
Tell me if you have a better idea.

Molrn commented on src/database/sqlite/event/event_store.py:
```diff
@@ -220,6 +223,74 @@ class StoredDisplayController:
+    active=True,
+    permissions={},
+    ip=None,
+)
```
these should be defined outside of the store (ex: `auth.__init__`)

pascalaubry replied:
Moved this code to a class method.

Molrn commented on src/plugins/chessevent/templates/chessevent_event_card_block.html:
```diff
@@ -1,4 +1,4 @@
-{% if admin_events_show_details %}
+{% if client.can_view_detailed_event_cards and admin_events_show_details %}
```
```suggestion
{% if admin_events_show_details %}
```
event_admin_show_details should always be false if not `client.can_view_detailed_event_cards`. This avoids propagation in the templates and potenitial errors. Same for every `client.can_view_detailed_{tab}_cards`

pascalaubry replied:
In  fact no: permissions can change while admin_xxx_show_details is and remains true, so both tests are needed.

pascalaubry committed:
Fixed can_set_specials_results_by_tournament_id

pascalaubry committed:
Fixed boards screen modal opening
Removed unused logs

Molrn commented:
# UI / integration feedback  

- suggestion: card containers for devices / accounts
- Tournament roles selection should be a multi-select (see prize criteria > Federation)
![image](https://github.com/user-attachments/assets/e31fc26c-61eb-4eb6-a7ce-d5173db07b78)
- The `Accès` menu should not appear if the mode does not allow it. In general, for modes, hiding the options should be prefered to disabling them
![image](https://github.com/user-attachments/assets/32886238-d93d-4be2-8b49-534fb57864d2)
- this should IMO be a select in an individual form, similar to prizes. When a prize category is created, it has no criteria with a warning on its card. You can then add criteria individually, with a select in a form modal showing a dynamic form (in this case: a tournament select + the exlanatory text for the permission)  
- There should not be elements displayed by default and instead a documentation text (like screens and prizes). The doc ending by `Create default devices to get started` with a button `Create default devices`
![image](https://github.com/user-attachments/assets/88ff619f-667f-4577-ae42-9a057fd2d07d)
- this should be something less techy and generic. The language should only be displayed on the modal. 
![image](https://github.com/user-attachments/assets/0b7b8848-8c5b-406c-9ccf-229a1ef47cca) suggestion:  ![image](https://github.com/user-attachments/assets/bd1ce066-bf53-430e-821a-f3c6c88a0f2c)

pascalaubry committed:
Restrained the private screens visualisation

pascalaubry committed:
Updated the documentation

pascalaubry committed:
Fixed typo

pascalaubry committed:
Fixed typo

pascalaubry committed:
Renamed computers to devices
Renamed ordinateur to appareil
Only kept 3 exec modes
Renamed mode to exec_mode (to explicitly differ from safety_mode)

pascalaubry committed:
Fixed FR translation

pascalaubry committed:
Fixed standard mode help text

pascalaubry committed:
Merged dev
Fixed Tim's comments (except one)

pascalaubry commented:
> * The `Accès` menu should not appear if the mode does not allow it. In general, for modes, hiding the options should be prefered to disabling them
>   ![image](https://private-user-images.githubusercontent.com/76754000/460706665-32886238-d93d-4be2-8b49-534fb57864d2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTE0MTEwNDYsIm5iZiI6MTc1MTQxMDc0NiwicGF0aCI6Ii83Njc1NDAwMC80NjA3MDY2NjUtMzI4ODYyMzgtZDkzZC00YmUyLThiNDktNTM0ZmI1Nzg2NGQyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzAxVDIyNTkwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIwZjFmNTYxMjM1YjI1Y2Y4MGFkMThiMzk0NTYwZmVjZmYyY2NkNGM0M2I0OTFjYTc4Y2JlYjEyZjQ4ODE5MzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.7S4B4NfKaZyBQKZfKi_yficH-t3jBV6gZ_kz7JLfWNE)

Done.

pascalaubry commented:
> * There should not be elements displayed by default and instead a documentation text (like screens and prizes). The doc ending by `Create default devices to get started` with a button `Create default devices`
>   ![image](https://private-user-images.githubusercontent.com/76754000/460710983-88ff619f-667f-4577-ae42-9a057fd2d07d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTE0MTEwNDYsIm5iZiI6MTc1MTQxMDc0NiwicGF0aCI6Ii83Njc1NDAwMC80NjA3MTA5ODMtODhmZjYxOWYtNjY3Zi00NTc3LWFlNDItOWEwNTdmZDJkMDdkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzAxVDIyNTkwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA5YTRiYjM4OTg0MDFlMTVmYmQ2OWIzOWQxNThlNDFmMDI5MWQ2NDU0ZjM1MmU2MTczMmRiMDJmYTdhZjVmNjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.QYiIBfpACK5ygp5NGGmv0-WnZCzldI_OVX1AtgZb2-g)

They are not elements displayed by default, they are the real permissions set in the mode (now they are shown only for `ExecMode.CUSTOM`)..

pascalaubry committed:
Default to the app config unless overridden
Move execution mode to lower down in the app and tournament config
6 to 3 modes
In the roles dialog, remove the tooltip
Hide the access menu unless in roles mode
Added MigrationDatabase._post_upgrade()

pascalaubry commented:
Very nice review, very complementary to Tim's.
Thank you very much.

Molrn commented on src/database/sqlite/migration_database.py:
```diff
@@ -126,6 +127,12 @@ def create(self):
+    def _post_migrate(self):
+        """Allow databases to do thing after the last migration,
+        see EventDatabase._post_migrate for an example."""
+        pass
```
That's a terrible idea, it will be executed at the end of every new migration. Currently, it would add your new devices over and over. Even with a check of only append if the ids exist, it is inefficient and a bad maintenance idea.  
If you need to add default objects, it has to be done in the migration, like the one row in the info table which is added in m001. Data insertion is also a valid migration operation, it's not just about the data structure.

pascalaubry replied:
You are absolutely right (forgot that it will be run at **every** migration).
I will restore what I did previously until we find a better way (let's talk about it).

pascalaubry committed:
Fixed multiple roles on accounts and devices tabs

pascalaubry committed:
Use fake accounts and devices for all modes (must still add real objects before editing them).

pascalaubry committed:
Merge branch 'dev' into feature/roles

pascalaubry committed:
Merge branch 'dev' into feature/roles

pascalaubry committed:
Create exec mode accounts and devices only when starting the editing of the permissions

timothyarmes committed:
Test fixes

pascalaubry committed:
Merge remote-tracking branch 'sharly-chess/feature/roles' into feature/roles

pascalaubry committed:
Fixed screen creation

pascalaubry committed:
Fixed the buttons of the account/device modals

pascalaubry committed:
Added a simple client tracker

timothyarmes committed:
Fix button icons

timothyarmes committed:
Fix tests

Molrn commented on src/web/templates/common/profile/profile_modal.html:
```diff
@@ -0,0 +1,221 @@
+    </div>
+</div>
+
+<script>
+    // We need to reload the page when the modal is hidden, because profile and language may have changed.
+    $(document).one('hide.bs.modal', function(e) {
+        e.preventDefault();
+        document.body.dispatchEvent(new CustomEvent('request_refresh'));
+    });
+</script>
```
requires the base page to have a `refresh_listener`, but the modal can be opened everywhere, even on pages without it. It therefore just prevents the default behavior and does not close the modal

timothyarmes committed:
Fix double click needed to log in and out

timothyarmes committed:
Role testing infrastructure

Amaras committed:
Improve security through password handling (#809)

* Add proper auth

* Add argron2 direct dependency

* Update m031 migration with better pw

Molrn committed:
Use entities for roles + Controller refactor + permissions multi-select form (#799)

* use entities for roles

* fix @abstractmethod order

* Full Role entity + Client

* Controller refactor + multi select permissions form

Molrn committed:
merge dev into feature/roles

Molrn committed:
fix tests

timothyarmes committed:
Fix tests

timothyarmes committed:
Anonymous and Spectator tests

timothyarmes committed:
Fix tests

timothyarmes committed:
Merge branch 'dev' into feature/roles

Amaras committed:
Merge branch 'dev' into feature/roles

pascalaubry committed:
Grant no roles by default to unknown devices in CUSTOM mode (#830)

pascalaubry committed:
Give roles real role names (not person names) (#833)

* Added missing i18n strings

* Give roles real role names (not person names)

* Fixed dst URLs

* Added Tim's suggestions

Molrn committed:
Make role permissions form dynamic (#835)

- display the help text in a tooltip
- disable the inherited roles with a `inherited` label
- only display the tournament selector if a tournament scope role is selected

pascalaubry committed:
Do not update i18n files for tests (#831)

* Do not update i18n files for tests

* Removed debug log

pascalaubry committed:
Use lists instead of dicts in client.py (#838)

pascalaubry committed:
Order roles (#839)

* Use lists instead of dicts in client.py

* Order roles

* Use the RoleManager to sort roles

Amaras committed:
Use a proper check to fix the IndexError (#843)

pascalaubry committed:
Allow devices/accounts to have no permissions (#840)

* Allow devices/accounts to have no permissions

* Made explicit the roles placeholder

pascalaubry committed:
Keep only two execution modes (standard/custom) (#841)

* Keep only two execution modes (standard/custom)

* Removed the constants from event_store.py

* Removed the default execution mode from config_tab.html

* Removed SharlyChessConfig dependency from migrations

* Removed class ExecMode

Amaras committed:
Fix the abstraction level of the Client class (#844)

Amaras committed:
fix typo

pascalaubry committed:
Merged dev

pascalaubry committed:
Fixed migration numbers

pascalaubry committed:
Added a first guard (#837)

* Added a first guard

* Renamed methods

pascalaubry committed:
Fixed 'CLEAR THE RESULT' button on user result modal (#850)

pascalaubry committed:
Fix Client.imputable_results_for_tournament() (#851)

* Fixed 'CLEAR THE RESULT' button on user result modal

* Fixed Client.imputable_results_for_tournament()

pascalaubry committed:
Merge branch 'dev' into feature/roles

pascalaubry committed:
Merged dev

pascalaubry committed:
Add guards on every user end-point (#852)

timothyarmes committed:
Merge branch 'dev' into feature/roles

timothyarmes committed:
Add a permission for setting illegal moves

timothyarmes committed:
Allow illegal moves entry from pairings tab

timothyarmes committed:
Fix admin account update

pascalaubry committed:
Merged dev (tests pass only once)

pascalaubry committed:
Merge branch 'dev' into feature/roles

pascalaubry committed:
Merged dev

pascalaubry committed:
Restrain the visibility of passed/coming events to administrators (#853)

* Fixed 'CLEAR THE RESULT' button on user result modal

* Fixed Client.imputable_results_for_tournament()

* Added guards on every user end-point

* Restrain the visibility of passed/coming events and archives

* Updated docs

* Fixed check-in test

* Fixed event_layout.html

* Fixed pairings modal (#856)

* Moved the methods out of the store (duplicated code)

* Fixed merge issue

pascalaubry committed:
Merge branch 'dev' into feature/roles

pascalaubry committed:
Updated migration numbering

pascalaubry committed:
Merged dev

pascalaubry committed:
Merged dev

pascalaubry committed:
Merge branch 'dev' into feature/roles

pascalaubry committed:
Merge branch 'dev' into feature/roles

pascalaubry committed:
Merged dev (Big Move)

pascalaubry committed:
Fixed duplicated {% endif %}

pascalaubry committed:
Added missing data-testid

pascalaubry committed:
Removed forgotten Access stuff

pascalaubry committed:
Removed forgotten Papi path

pascalaubry committed:
Added missing strings

pascalaubry committed:
Fixed import order

pascalaubry committed:
Fixed role_test_events()

pascalaubry committed:
Fixed role_test_tournament()

timothyarmes committed:
Fix some tests

timothyarmes committed:
Don't log stack exceptions for Permission errors

timothyarmes committed:
Fix tests (out of date cached pages were returned)

timothyarmes committed:
Remove unnecessary polling

pascalaubry committed:
Merged dev

pascalaubry committed:
Merged dev

pascalaubry committed:
Applied #941 to the pairings tab

pascalaubry committed:
Merged dev (3.0.0b1)

pascalaubry committed:
Merged dev (Event unique ID)

pascalaubry committed:
Merge branch 'dev' into feature/roles

pascalaubry committed:
Fixed i18n strings

pascalaubry committed:
Generate roles inheritance doc

pascalaubry committed:
Merged dev

pascalaubry committed:
Merged dev

pascalaubry committed:
Merge branch 'feature/roles' into feature/roles-doc

pascalaubry committed:
generate permissions doc

pascalaubry committed:
Merge branch 'dev' into feature/roles

pascalaubry committed:
Merge branch 'feature/roles' into feature/roles-doc

pascalaubry committed:
Merged dev

pascalaubry committed:
Merged dev

pascalaubry committed:
Merged dev

pascalaubry committed:
Merge branch 'dev' into feature/roles

pascalaubry committed:
Merge branch 'dev' into feature/roles

pascalaubry committed:
Merge branch 'feature/roles' into feature/roles-doc

pascalaubry committed:
Fixed previous merge

pascalaubry committed:
Fixed translations

pascalaubry committed:
Force the account passwords not to be null

pascalaubry committed:
Force the account passwords not to be null
Fixed manageable_roles() not to allow ORG to manage all the CA sub-roles
Made PAI inherit from SEC
Changed the scope of DCA to Tournament

pascalaubry committed:
Updated the roles documentation

pascalaubry committed:
Merged dev (#965)

pascalaubry committed:
Fixed NOT NULL contraint for anonymous accounts

pascalaubry committed:
Merged dev (#970, #971 and #972)

pascalaubry committed:
Merged dev (#973)

pascalaubry committed:
Merged dev (#969)

pascalaubry committed:
Merge branch 'dev' into feature/roles

pascalaubry committed:
Merged dev

pascalaubry committed:
Merged dev

pascalaubry committed:
Merged dev

pascalaubry committed:
Merged dev

pascalaubry committed:
Merge branch 'dev' into feature/roles

pascalaubry committed:
Merge branch 'dev' into feature/roles

pascalaubry committed:
Merge branch 'dev' into feature/roles

pascalaubry committed:
Merged dev

pascalaubry committed:
Merged dev

pascalaubry committed:
Merged dev

pascalaubry committed:
Merge branch 'dev' into feature/roles

pascalaubry committed:
Fixed migration numbers

pascalaubry committed:
Merged dev

pascalaubry committed:
Merged dev

pascalaubry committed:
Merge branch 'dev' into feature/roles

pascalaubry committed:
Merged dev

pascalaubry committed:
Merged dev (tests fail on uniq-id problems)

pascalaubry committed:
Merge branch 'dev' into feature/roles

timothyarmes committed:
Fix tests

pascalaubry committed:
Merge remote-tracking branch 'sharly-chess/feature/roles' into feature/roles

pascalaubry committed:
Merged dev (tests all pass except test_results_entry_screen)
Fixed log trace on result input

pascalaubry committed:
Merge branch 'dev' into feature/roles

pascalaubry committed:
Merge branch 'dev' into feature/roles

pascalaubry committed:
Merged dev (tests pass, web UI fails with database is locked)

pascalaubry committed:
Merge branch 'dev' into feature/roles

pascalaubry committed:
Update usages of Tournament.uniq_id

pascalaubry committed:
Added option --port to force the port use by the web server

pascalaubry committed:
Fixed translations

pascalaubry committed:
Merge branch 'domloup-20250830/dev' into feature/roles

pascalaubry committed:
Fixed DisplayController() and RotatorController() calls

pascalaubry committed:
Set scope Tournament to role DeputyChiefArbitrationRole

pascalaubry committed:
Add missing methods has_xxx_scope()

pascalaubry committed:
Fix the title of input/boards sets

pascalaubry committed:
Fix mouse pointer and check-in checkboxes on input/boards sets

pascalaubry committed:
Fix "Close check-in" button tooltip

pascalaubry committed:
Fix "Close check-in" button tooltip alignment

pascalaubry committed:
Fix "Search for player" input tooltip alignment

pascalaubry committed:
Fix the title of input/boards sets
Fix mouse pointer and check-in checkboxes on input/boards sets
Fix "Close check-in" button tooltip
Fix "Search for player" input tooltip alignment

pascalaubry committed:
Fix the display of permissions

pascalaubry committed:
Fix accounts/devices display and editing

pascalaubry committed:
Fix UNPAIR button style

pascalaubry committed:
Fix IP pattern

pascalaubry committed:
Fix FR translation

pascalaubry committed:
Merge branch 'dev' into feature/roles

pascalaubry committed:
Merged dev

pascalaubry committed:
Fixed missing parenthesis

pascalaubry committed:
Merged dev

pascalaubry committed:
Merged dev

timothyarmes reviewed (approved):
I"ve been through it.  It's basically impossible to review, there are so many changes.  In particular to the templates.
I just want to get it merged so that we can move forwards with refactoring and writing tests

Molrn reviewed (approved):
Re-read, test needs fixing before merging but otherwise we can merge

pascalaubry committed:
Merged dev

pascalaubry committed:
Merged #1081 to fix test failure on https://github.com/Sharly-Chess/sharly-chess/actions/runs/17402909617/job/49400005231
